aws/resource_aws_iot_policy_test.go

```diff
@@ -0,0 +1,109 @@
+package aws
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/service/iot"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+	"regexp"
+)
+
+func TestAccAWSIoTPolicy_basic(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAWSIoTPolicyDestroy_basic,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAccAWSIoTPolicy_basic,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSIoTPolicyExists_basic("aws_iot_policy.pubsub"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccAWSIoTPolicy_invalidJson(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAWSIoTPolicyDestroy_basic,
+		Steps: []resource.TestStep{
+			{
+				Config:      testAccAWSIoTPolicyInvalidJsonConfig,
+				ExpectError: regexp.MustCompile("MalformedPolicyException.*"),
+			},
+		},
+	})
+}
+
+func testAccCheckAWSIoTPolicyDestroy_basic(s *terraform.State) error {
+	conn := testAccProvider.Meta().(*AWSClient).iotconn
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "aws_iot_policy" {
+			continue
+		}
+
+		out, err := conn.ListPolicies(&iot.ListPoliciesInput{})
```

Comment:
Wouldn't it be easier to call `GetPolicy` here as we only care about the single policy we created as part of the test in this context?